Source/core/NetworkInfo.cpp

```diff
@@ -1258,35 +1232,109 @@ namespace Core {
             }
         }
 
+        void AddEventObserver(INetworkNotification* callback)
+        {
+            _channel->AddEventObserver(callback);
+        }
+
+        void RemoveEventObserver(INetworkNotification* callback)
+        {
+            _channel->RemoveEventObserver(callback);
+        }
     private:
-        ProxyType<Channel> _channel;
+        ProxyType<SocketIPNetworks> _channel;
         std::map<uint32_t, Network> _networks;
         Network _invalidNetwork;
     };
 
+    // TODO: Redesign in asynchronous way
     uint32_t IPNetworks::Network::Add(const IPNode& address)
     {
         IPAddressModifyType<true> modifier(*this, address);
 
-        return (_channel->Exchange(modifier, modifier));
+#ifdef __DEBUG__
+    const uint32_t waitTime = Core::infinite;
+#else
+    const uint32_t waitTime = 2000;
+#endif
+
+        return (_channel->Exchange(modifier, modifier,  waitTime));
     }
 
     uint32_t IPNetworks::Network::Delete(const IPNode& address)
     {
         IPAddressModifyType<false> modifier(*this, address);
 
-        return (_channel->Exchange(modifier, modifier));
+#ifdef __DEBUG__
+    const uint32_t waitTime = Core::infinite;
+#else
+    const uint32_t waitTime = 2000;
+#endif
+
+        return (_channel->Exchange(modifier, modifier, waitTime));
     }
 
     uint32_t IPNetworks::Network::Gateway(const IPNode& network, const NodeId& gateway)
     {
         IPRouteModifyType<true> modifier(*this, network, gateway);
 
-        return (_channel->Exchange(modifier, modifier));
+#ifdef __DEBUG__
```

Comment:
Move this waitTime definition used (and now copied multiple times :-) ) to a single class static constexpr...

Reply:
2 Changes were made. The Size() -> PayloadSize() you mentioned and Data() -> template<> Payload(), however, both changes were made in Frames class that is used to parse Netlink messages and wasn't used anywhere else (event in code you provided in conversation). 

These changes are obviously not absolutely required, however it was confusing at times whether Data() and Size() were referring to just payload or the whole frame.

Reply:
Actually after reviewing one more thing changed in Network (Event class), however, it came mostly from cutting changes into this smaller PR and I will undo it